### PR TITLE
V3 Conformance + `zos-node-accessor` update

### PR DESCRIPTION
**Pull request by zFernand0**
**What It Does**

- Address most/all breaking changes from imperative and other SDKs
- Update `zos-node-accessor`  to v2

**How to Test**

Build and run the plug-in 😋 

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [ ] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

More changes to come 😋

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -87,26 +87,27 @@ Before you install the plug-in, meet the following requirements:
```
Comment (anaxceron) [inline README.md:87]: Edits for other parts of the ReadMe:

- Line 3: data sets should be two words
- Line 9: trim the sentence: "You can use the plug-in to perform various tasks:"
- Line 11: data sets two words; also ".. download **a** z/OS data set ..."
- Line 12: " ... to **a** z/OS **data set** or ..."
- Line 13: "... or download a job status and job spool files"
- Line 14: "Delete **a** z/OS **data set** ..."
- Line 15: "Submit **a** job from **a data set** ..."
- Line 16: "Make **a** USS directory"
- Line 17: "Allocate **a data set**"
- Line 19: lowercase: Software requirements
- Line 27: "**Ensure the** z/OS FTP service can be accessed remotely. Connect z/OS with the FTP client and run **the** command `rstat` in FTP client to check **that** option JESINTERFACELevel is set to 2."
- Line 29: Lowercase: Build the plug-in from source
- Line 33: What happens if this is not the first time? Should we add instructions for that scenario?
- Line 47: Same as Line 33.
- Line 86: Remove "Tip:." Tips and troubleshooting are two different things. This is troubleshooting, not a tip.

File: README.md [hunk 2]
```diff
@@ -87,26 +87,27 @@ Before you install the plug-in, meet the following requirements:
 
 5.  [Create a user profile](#create-a-user-profile).
 
-## Create a User Profile
+## Creating a user profile
 
-You can create a `zftp` user profile to avoid typing your connection details on every command. A `zftp` profile contains the host, port, username, and password for the z/OS you will connect. You can create multiple profiles and switch between them as needed.
+After you install the plug-in, you create an FTP profile. An FTP profile is recommended to issue commands to via FTP. FTP profiles contain your host, port, user name, and password for connect to z/OS using FTP. You can create multiple profiles and switch between them as needed.
```
Comment (anaxceron) [inline README.md:92]: Change to: "After you install the plug-in, create an FTP profile"

File: README.md [hunk 3]
```diff
@@ -87,26 +87,27 @@ Before you install the plug-in, meet the following requirements:
...
+4. If using a insecure connection, set the secureFtp value to false:
 
-```
-zowe profiles create zftp -h
-```
+    ```
+    zowe config set profiles.zftp.properties.secureFtp false
+    ```
 
 ## Running tests
```
Comment (anaxceron) [inline README.md:112]: More notes:
Line 121: Not clear what the user needs to do once the file is copied. Should the user rename the copied file to `custom_properties.yaml`?
Line 123: change the tense: "... where the system test will be **ran** for the first time ... ." Also, data set it two words. Also: " ... **issue** the following command ..."
Line 127: Indent this content and split it up into separate lines, one for localfile and one for dataset
Line 139: lowercase: Uninstall the plug-in
Line 146: state what the actual "product" is
Line 162: we should explain why we're including this example script. It's not clear as it is now.

**Review by anaxceron — APPROVED** [review]
Your particular changes look pretty good. The rest of the ReadMe should be edited. I left comments.

File: src/api/CoreUtils.ts [hunk 4]
```diff
@@ -82,23 +64,24 @@ export class CoreUtils {
...
             // z/osmf so that users who use the list ds command in main
             // zowe can use the same filtering options
             this.log.trace("Remapping key for data set to match core CLI. Old key '%s' New key '%s'", key, key.toLowerCase());
             result[key.toLowerCase()] = obj[key];
         }
-        return result;
+        return result as typeof obj;
     }
 
     public static async getProfileMeta(): Promise<ICommandProfileTypeConfiguration[]> {
```
Comment (gejohnston) [inline src/api/CoreUtils.ts:87]: You might want to check if function getProfileMeta is used anywhere. If it is trying to read V1 profile meta file information, this function may no longer be necessary.

**Review by gejohnston — APPROVED** [review]
I left one comment that you could investigate. Worst case, leaving function getProfileMeta in the code is probably harmless.

File: README.md (hunk 1 above)
Comment (zFernand0) [reply README.md:87]: > Line 27:

Reworded:
"Ensure the z/OS FTP service can be accessed remotely by connecting to z/OS from an FTP client. From the FTP client, run the command `rstat` to verify that the option `JESINTERFACELevel` is set to 2."


> Line 33: What happens if this is not the first time? Should we add instructions for that scenario?

Good point, we should advice people to run `npm install` after downloading  the source from GitHub 😋 

> Line 47

That sentence doesn't apply anymore, so I removed it 😋 

> Line 86:

Reworded:
"**Note:** When an unsuccessful message displays, you can troubleshoot the installation by addressing the issues that the message describes. For more details, you can review the logs located in the directory where you installed the Zowe CLI."

File: README.md (hunk 3 above)
Comment (zFernand0) [reply README.md:112]: > Line 162

Added some comments to the script and slightly updated the intro to the script 😋

File: src/api/CoreUtils.ts (hunk 4 above)
Comment (zFernand0) [reply src/api/CoreUtils.ts]: Renamed to `getProfileSchema` and removed any references to V1 profiles 😋 
Good catch 😋

**Review by anaxceron — APPROVED** [review]
great!

**Review by traeok — CHANGES_REQUESTED** [review]
Thanks for all of your work on this!
I noticed that a couple of the unit tests are failing on Windows - the failures seem related to path formatting. Will approve once addressed.

**Pull request merged**